objectstore-client/src/client.rs

```diff
@@ -115,25 +131,53 @@ impl fmt::Debug for Client {
             .field("usecase", &self.usecase)
             .field("scope", &self.scope)
             .field("default_compression", &self.default_compression)
+            .field(
+                "enable_distributed_tracing",
+                &self.enable_distributed_tracing,
+            )
             .finish()
     }
 }
 
 /// The type of [`Stream`](futures_util::Stream) to be used for a PUT request.
 pub type ClientStream = BoxStream<'static, io::Result<Bytes>>;
 
+/// HTTP header that Sentry uses to propagate distributed tracing IDs.
+const SENTRY_TRACE_HEADER: &str = "sentry-trace";
+
 impl Client {
+    pub(crate) fn request<U: reqwest::IntoUrl>(
+        &self,
+        method: reqwest::Method,
+        uri: U,
+    ) -> reqwest::RequestBuilder {
+        let mut builder = self.http.request(method, uri).query(&[
+            (PARAM_SCOPE, self.scope.as_ref()),
+            (PARAM_USECASE, self.usecase.as_ref()),
+        ]);
+
+        if self.enable_distributed_tracing {
+            // If the Sentry SDK has been initialized and tracing is in progress,
+            // this will set the header Sentry uses to propagate distributed
+            // tracing context. Otherwise, nothing happens.
+            if let Some(trace_id) = sentry::configure_scope(|scope| {
+                scope
+                    .get_span()
+                    .map(|span| span.get_trace_context().trace_id)
+            }) {
+                builder = builder.header(SENTRY_TRACE_HEADER, trace_id.to_string());
```

Comment:
There is [`Scope::iter_trace_propagation_headers`](https://docs.rs/sentry/latest/sentry/struct.Scope.html#method.iter_trace_propagation_headers) for this purpose, so you don’t need to add a hardcoded header, but the SDK will expose any headers in whichever format it wants, in a forward compatible way.